docs/DeveloperGuide.md

```diff
@@ -161,6 +163,67 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+## Optional Fields
+
+The transformation of certain fields in the Internship class to optional types is a crucial step in enhancing the flexibility and robustness of the internship application.
+This report outlines the technical details of the implementation, including the different states, usage scenarios, design considerations, alternatives, and their implications.
+
+<img src="images/OptionalFeatures.png" width="280" />
+
+#### Implementation Details (How Is It Implemented):
+
+**Field Conversion:**
+Fields such as location and role have been converted to optional types (where default values have been set should the user not input any values for them)
+This transformation allows these fields to be nullable, providing flexibility in representing internship data.
+
+**Constructor Modification:**
+The constructor of the Internship class now accepts optional parameters for the transformed fields, ensuring compatibility with the new optional field structure.
```

Comment:
technically the constructor does not accept optional parameters
The parser accepts optional fields and sets default values in the constructor